lib/References/Resolver.php

```diff
@@ -15,6 +16,19 @@ public function resolve(
         string $data,
         array $attributes = []
     ): ?ResolvedReference {
+        $eventManager = $environment->getConfiguration()->getEventManager();
+
+        $preReferenceResolvedEvent = new PreReferenceResolvedEvent($environment, $data, $attributes);
+
+        $eventManager->dispatchEvent(
+            PreReferenceResolvedEvent::PRE_REFERENCED_RESOVED,
+            $preReferenceResolvedEvent
+        );
+
+        if ($preReferenceResolvedEvent->getResolvedReference() !== null) {
+            return $preReferenceResolvedEvent->getResolvedReference();
```

Comment:
You could write an integration test for `Resolver`, with real `Environment`, `Configuration` and `EventManager` objects, and register a dummy event listener in it, that would change the url to something hardcoded, and check that the url is indeed changed.

Reply:
done